heatintro.md

```diff
@@ -1,85 +1,142 @@
-    - cooldown: minimum amount of time (in seconds) between allowable executions of this policy.
-    - group_id: ID of the group that this policy will affect
-    - name: Convenient name
-
-    Attributes: (outputs)
-    - alarm_url: the url which triggers the policy
-
-
-*OS::Monasca::Notification*
-    Properties:
-    - address: Address of the notification. 
+---
+## OS::Heat::AutoScalingGroup
+Properties:
+  - **name:** Convenient name.
+  - **max_size:** Maximum size of the group.
+  - **min_size:** Minimum size of the group.
+  - **cooldown:** minimum time before *any* autoscaling related operation.
+  - **resources:** set of resource that should be scaled up or down.
+
+Example:
+
+    group:
+      type: OS::Heat::AutoScalingGroup
+      properties:
+        ... specify propoerties from above ..
+        resource:
+          type: OS::Nova::Server
+          properties:
+            flavor: { get_param: flavor }
+            image: { get_param: image }
+            # pass scale_group as metadata,
+            # monasca will store this as vm metadata and
+            # later can be used to group vms together from
+            # same stack
+            metadata: {"scale_group": {get_param: "OS::stack_id"}}
+            networks:
+            - network: private
+            security_groups:
+            - testvm
+
+---
+## OS::Heat::ScalingPolicy
+Properties:
+  - **change**: a number that has an effect based on change_type.
+  - **change_type:** (meaning of "change")
+  	- "change_in_capacity"
+    - "percentage_change_in_capacity", 
+    - "exact_capacity" 
+  - **cooldown:** minimum amount of time (in seconds) between allowable executions of this policy.
+  - **group_id:** ID of the group that this policy will affect
+  - **name:** Convenient name
+
+Attributes: (outputs)
+  - **alarm_url**: the url which triggers the policy
+
+Example:
+
+    policy_for_autoscaling_group:
+      type: OS::Heat::ScalingPolicy
+      properties:
+        adjustment_type: change_in_capacity
+        auto_scaling_group_id: { get_resource: group }
+        cooldown: 60
+        scaling_adjustment: -1
+
+---
+
+## OS::Monasca::Notification
+Properties:
+  - **address**: Address of the notification. 
         - email address
         - url
         - service key based on notification type.
-    - type: Type of the notification.
+  - **type**: Type of the notification.
         - email
         - webhook
         - pagerduty
-exameple:
-  up_notification:
-    type: OS::Monasca::Notification
+     
+Example:
+
+	notification_object:
+      type: OS::Monasca::Notification
+      properties:
+        type: webhook  # we want the notification to trigger the policy via a url
+        address: {get_attr: [some_policy, alarm_url]} # this get the alarm_url to trigger the policy
+
+---
+
+## **OS::Monasca::AlarmDefinition**
+Properties:
+- **expression**: Expression of the alarm to evaluate.
+	
+    eg: `avg(cpu.utilization_perc{group_by=something}) > 50 times 3`
```

Comment:
I think we will want to break down how expressions work, at least this example.  And likely link to the Monasca doc on expressions.

avg = average the values
cpu.utilization_perc{group_by=something} = the CPU utilization for metrics reported with a dimension that matches the given value.
> 50 = a value comparison, setting a threshold for the expression
times 3 = number of times the threshold should be passed before triggering the alarm.  This is important to avoid thrashing around and to 'smooth out' small spikes. (Don't confuse with "50 * 3")

Reply:
There may be some questions about how often we sample and how quickly we can autoscale. We want to be prepared to discuss the tradeoff between quick response and avoiding thrashing around. Part of the answer is choosing threshold values that occur before it becomes a problem (don't wait for 100% usage).

Reply:
I've been trying to think of other expressions/alarms that might be useful.  So far I've considered the host_alive metric which could be used to trigger a self-healing behavior for scale up (though I'd have to think about how that would be reported... maybe a sum() < 3 and sum > 4, though there are other ways to achieve that) or maybe avg(net.bytes_in_sec), though defining a threshold for that requires knowing more about the network provisioned and what an acceptable load is.  Those might be good for an 'extra credit' section.